enterprise/admin/views.py

```diff
@@ -62,20 +64,56 @@ def _build_context(self, request, customer_uuid):
         """
         # TODO: pylint acts stupid - find a way around it without suppressing
         enterprise_customer = EnterpriseCustomer.objects.get(uuid=customer_uuid)  # pylint: disable=no-member
-        linked_learners = EnterpriseCustomerUser.objects.filter(enterprise_customer__uuid=enterprise_customer.uuid)
-        pending_linked_learners = PendingEnterpriseCustomerUser.objects.filter(
-            enterprise_customer__uuid=enterprise_customer.uuid
-        )
+        search_keyword = self.get_search_keyword(request)
+        linked_learners = self.get_enterprise_customer_user_queryset(search_keyword, customer_uuid)
+        pending_linked_learners = self.get_pending_queryset(search_keyword, customer_uuid)
 
         context = {
             self.ContextParameters.ENTERPRISE_CUSTOMER: enterprise_customer,
             self.ContextParameters.PENDING_LEARNERS: pending_linked_learners,
             self.ContextParameters.LEARNERS: linked_learners,
+            self.ContextParameters.SEARCH_KEYWORD: search_keyword or '',
         }
         context.update(admin.site.each_context(request))
         context.update(self._build_admin_context(request, enterprise_customer))
         return context
 
+    def get_search_keyword(self, request):
+        """
+        Retrieve the search querystring from the GET parameters.
+        """
+        return request.GET.get('q', None)
+
+    def get_enterprise_customer_user_queryset(self, search_keyword, customer_uuid):
+        """
+        Get the list of EnterpriseCustomerUsers we want to render.
+        """
+        learners = EnterpriseCustomerUser.objects.filter(enterprise_customer__uuid=customer_uuid)
+
+        if search_keyword is not None:
+            user_ids = learners.values_list('user_id', flat=True)
+            allowed_users = User.objects.filter(
```

Comment:
This might be just my nitpicky nature, but I'm wondering about the choice of "allowed" here, versus something more along the lines of "matching" -- "allowed" makes me think that there's some sort of AuthZ going on?